src/emit/mod.rs

```diff
@@ -253,6 +267,7 @@ pub trait TargetRuntime<'a> {
     ) -> BasicValueEnum<'a>;
 
     /// Calls constructor
+    #[allow(clippy::too_many_arguments)]
```

Comment:
Clippy is right.

Reply:
You don't need to fix this. This should go in a larger refactoring.

Reply:
The salt/seed/value create contract arguments should be collected in a struct and passed as one

Reply:
@salaheldinsoliman thanks for doing that, I actually meant to say this can be done in another PR, but  looks great - thank you! :pray:

Reply:
@seanyoung no worries